store/tikv/2pc.go

```diff
@@ -135,6 +147,27 @@ func newTwoPhaseCommitter(txn *tikvTxn, connID uint64) (*twoPhaseCommitter, erro
 	if len(keys) == 0 {
 		return nil, nil
 	}
+
+	for _, pair := range txn.contracts {
+		mutation, ok := mutations[string(pair.key)]
+		if !ok {
+			log.Info(" something is wrong... contract exists but no mutation?", pair)
+			continue
+		}
+		// Only apply the first contract!
+		if mutation.Precondition != nil {
+			continue
+		}
+		mutation.hasContract = true
+		mutation.Precondition = &pb.Precondition{}
+		switch pair.contract {
+		case kv.MustExist:
+			mutation.Precondition.MustExist = true
+		case kv.MustNotExist:
+			mutation.Precondition.ShouldNotExist = true
```

Comment:
```suggestion
			mutation.Precondition.MustNotExist = true
```
Should we use an uniform name for it? Both `should` or `must` are OK for me.

Reply:
This is defined in kvproto, not in TiDB
And the name is proposed by @zhangjinpeng1987

Reply:
I do agree with @jackysp

Reply:
https://github.com/pingcap/kvproto/pull/353
Add Insert and Update operation...